sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1976,11 +1977,14 @@ object SQLConf {
     .stringConf
     .createOptional
 
-  val V2_SESSION_CATALOG = buildConf("spark.sql.catalog.session")
-      .doc("A catalog implementation that will be used in place of the Spark built-in session " +
-        "catalog for v2 operations. The implementation may extend `CatalogExtension` to be " +
-        "passed the Spark built-in session catalog, so that it may delegate calls to the " +
-        "built-in session catalog.")
+  val V2_SESSION_CATALOG_IMPLEMENTATION =
+    buildConf(s"spark.sql.catalog.$SESSION_CATALOG_NAME")
+      .doc("A catalog implementation that will be used as the v2 interface to Spark's built-in " +
+        s"v1 catalog: $SESSION_CATALOG_NAME. This catalog shares its identifier namespace with " +
+        s"the $SESSION_CATALOG_NAME and must be consistent with it; for example, if a table can " +
+        s"be loaded by the $SESSION_CATALOG_NAME, this catalog must also return the table " +
+        s"metadata. To delegate operations to the $SESSION_CATALOG_NAME, implementations can " +
+        "extend 'CatalogExtension'.")
```

Comment:
Thanks for pinging me. Almost missing this one.

Does it mean, any implementation configured here, must be the v2 interface to built-in v1 catalog? Sounds like it can delegate or not delegate operations to $SESSION_CATALOG_NAME? If it does not, is it still be the v2 interface to the built-in v1 catalog?

A bit confused here.

Reply:
It must be a v2 interface, but we can't fully control how it is implemented. We expect it to delegate to the SESSION_CATALOG_NAME, but there is no way to guarantee it.

If the implementation doesn't delegate, then the behavior is undefined.